src/duckling/canvas/map-editor.component.ts

```diff
@@ -119,16 +122,68 @@ export class MapEditorComponent implements AfterViewInit {
 
         this._entitySystemService.entitySystem
             .map(this._entityDrawerService.getSystemMapper())
-            .subscribe((stage) => {
-                this._entitiesDisplayObject = stage;
-                this.canvasDisplayObject = this.buildCanvasDisplayObject();
+            .subscribe((drawnConstructs) => {
+                this.entitiesDrawnConstructsChanged(drawnConstructs);
             });
 
         this._assetService.assetLoaded.subscribe(() => {
-            let stage = this._entityDrawerService.getSystemMapper()(this._entitySystemService.entitySystem.value);
-            this._entitiesDisplayObject = stage;
-            this.canvasDisplayObject = this.buildCanvasDisplayObject();
+            let drawnConstructs = this._entityDrawerService.getSystemMapper()(this._entitySystemService.entitySystem.value);
+            this.entitiesDrawnConstructsChanged(drawnConstructs);
         });
+
+        setInterval(() => this.drawFrame(), this._frameRate);
+    }
+
+    private entitiesDrawnConstructsChanged(newDrawnConstructs : DrawnConstruct[]) {
+        this._lastDrawnConstructs = newDrawnConstructs;
+        this.createEntitiesDisplayObject(newDrawnConstructs)
+    }
+
+    private drawFrame() {
+        this._totalMillis += this._frameRate;
+        this.createEntitiesDisplayObject(this._lastDrawnConstructs);
+    }
+
+    private createEntitiesDisplayObject(entitiesDrawnConstructs : DrawnConstruct[]) {
```

Comment:
I think you should extract the createEntitiesDisplayOBject, drawConstructToDisplayObject, and determineAnimiationDisplayObject into their own functions or service.